packages/oats-runtime/src/make.ts

```diff
@@ -428,7 +473,11 @@ export function makeObject<
       }
       result[index] = propResult.success();
     }
-    return Make.ok(result);
+    const oldType = getType(value);
```

Comment:
in case we are in a nested object under an allOf the object we get as input may have been made already and has a reflection type. We merge this input value type with the current type so that the nested objects have right reflection types.

Alternative way would be to do recursively merge types in the makeAllOf handler. Don' know which is nicer

Reply:
This seemed fine to me.